packages/zoe/src/contracts/coveredCall-durable.js

```diff
@@ -0,0 +1,151 @@
+ * `give` and `want` with any keywords. Any number of assets of
+ * different brands can be escrowed under different keywords. The
+ * proposal must have an exit record with the key "afterDeadline":
+ * {
+ *    give: { ... },
+ *    want: { ... },
+ *    exit: {
+ *      afterDeadline: { deadline: time, timer: myTimer }
+ *    },
+ * }
+ *
+ * This deadline serves as the expiration date for the covered call
+ * option. After this deadline, if the option has not been exercised,
+ * the underlying assets are automatically paid out to the creator of
+ * the contract as a refund.
+ *
+ * After the owner of the digital assets escrows the assets in the
+ * initial offer, they receive a seat. The payout for this seat will
+ * either be a refund of the underlying assets (as mentioned above) or
+ * payments in the amount of the strike price. Zoe's enforcement of
+ * offer safety guarantees that the payout is either a refund or
+ * payments in the amount of the strike price, regardless of whether
+ * this contract is buggy.
+ *
+ * The offerResult of this initial seat resolves to the call option
+ * itself: an inspectable invitation to buy the underlying assets. The
+ * call option invitation has this additional information in the
+ * value: {expirationDate, timeAuthority, underlyingAssets,
+ * strikePrice }
+ *
+ * The invitation itself can be traded as a valuable digital asset: a
+ * covered call option.
+ *
+ * The recipient of a covered call option (buying it on some other
+ * exchange or through some other trading contract) can exercise the
+ * option before the deadline by using the option as an invitation to
+ * this contract, paying the strike price and receiving the underlying
+ * assets. The recipient of a covered call option can use whatever
+ * keywords they wish, as long as they give the strike price as
+ * specified in the invitation value, and want the underlying assets
+ * exactly.
+ *
+ * @param {import('@agoric/vat-data').Baggage} installationBaggage
+ * @param {ZCF} zcf passed in setupInstallation, but not completely initialized
+ *    until setupInstance. Passed early because it will often need to be
+ *    captured lexically by durable kinds.
+ */
+const setupInstallation = (installationBaggage, zcf) => {
+  const instanceBaggage = provideDurableMapStore(
+    installationBaggage,
+    'instance',
+  );
+
+  const makeExerciserKindHandle = provideKindHandle(
+    instanceBaggage,
+    'makeExerciserKindHandle',
+  );
+
+  const makeExerciser = defineDurableKind(
+    makeExerciserKindHandle,
+    sellSeat => ({ sellSeat }),
+    {
+      handle: ({ state: { sellSeat } }, buySeat) => {
+        assert(!sellSeat.hasExited(), sellSeatExpiredMsg);
+        try {
+          swapExact(zcf, sellSeat, buySeat);
+        } catch (err) {
+          console.log(
+            'Swap failed. Please make sure your offer has the same underlyingAssets and strikePrice as specified in the invitation details. The keywords should not matter.',
```

Comment:
As long as we're also sending a diagnostic to the log right here, we may as well also include the error we're about to rethrow.

```suggestion
            'Swap failed. Please make sure your offer has the same underlyingAssets and strikePrice as specified in the invitation details. The keywords should not matter.',
            err,
```

Reply:
done